### PR TITLE
Expose helper APIS for GetLower/Upper and WithLower/Upper

### PR DESCRIPTION
**Pull request by tannergooding**
This ensures the GetLower, GetUpper, WithLower, and WithUpper APIs are intrinsic and have appropriate helpers carried through to lowering.

This should minimally improve codegen as we no longer have to worry about the inliner but it should also make it simpler for certain AVX-512 optimizations to light up, such as `vector512.GetLower().GetLower()`.

### CODE REVIEW

**Comment by tannergooding** [comment]
WASM failures are https://github.com/dotnet/runtime/issues/83986

**Comment by tannergooding** [comment]
CC. @dotnet/jit-contrib, @dotnet/avx512-contrib

**Comment by tannergooding** [comment]
[Diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=221829&view=ms.vss-build-web.run-extensions-tab)

Some small improvement to Arm64 as we're able to optimize the `Vector128.GetUpper`:
```diff
-            movi    v17.4s, #0
-            ext     v16.16b, v16.16b, v17.16b, #8
+            ext     v16.16b, v16.16b, v16.16b, #8
```

No real diffs on x64. For both x64 and Arm64 it being 1 node with 1 (Get) or 2 (With) inputs all the way to lowering/codegen allows the JIT to do some minor transformations. In the case of Linux x64, there are a couple places where we get 1 memory read instead of 2. In the case of Arm64 we sometimes see the `ext` reordered closer to its actual usage site. These are most often no change in assembly size when this happens.

**Review by BruceForstall — APPROVED** [review]

File: src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector128.cs [hunk 1]
```diff
@@ -3168,22 +3162,16 @@ public static Vector128<T> WithLower<T>(this Vector128<T> vector, Vector64<T> va
         /// <param name="value">The value of the upper 64-bits as a <see cref="Vector64{T}" />.</param>
         /// <returns>A new <see cref="Vector128{T}" /> with the upper 64-bits set to <paramref name="value" /> and the lower 64-bits set to the same value as that in <paramref name="vector" />.</returns>
         /// <exception cref="NotSupportedException">The type of <paramref name="vector" /> (<typeparamref name="T" />) is not supported.</exception>
+        [Intrinsic]
```
[the remark this answers is not in the document]
Comment (tannergooding) [reply src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector128.cs:3165]: Will fix these in a follow up PR.

[the remark this answers is not in the document]
Comment (tannergooding) [reply src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector128.cs:3165]: https://github.com/dotnet/runtime/pull/84113

**Pull request merged**